x-pack/plugins/fleet/server/services/actions/types.ts

```diff
@@ -104,7 +104,7 @@ export interface FleetActionRequest extends CommonFleetActionResultDocFields {
     signature: string;
   };
   start_time?: string;
-  timeout: number;
+  timeout?: number;
```

Comment:
Have to make this optional as `osquery` actions don't have a timeout for fleet actions. @patrykkopycinski @tomsonpl @juliaElastic

Reply:
Hmm, interesting. Do you happen to know what `timeout` does? Isn't the behavior the same as based on `expiration` ?

Reply:
`expiration` is the time after which the action would not be a _delivered_ to the endpoint. `timeout` on the other hand is the period that the agent waits for a response from the endpoint. I believe it defaults to [1 minute](https://github.com/elastic/beats/blob/a55ae5857c0d80e0834c9f3e98fd1b833675f12c/x-pack/elastic-agent/pkg/agent/application/pipeline/actions/handlers/handler_action_application.go#L20) if not provided. @aleksmaus correct me if I'm wrong.